ui/src/app/views/concept-homepage/component.ts

```diff
@@ -69,13 +70,15 @@ export class ConceptHomepageComponent implements OnInit {
 
   // Maps Domain Name with number of concept selected
   selectedConceptDomainMap: Map<String, number>  = new Map<string, number>();
-  concepts: Array<Concept> = [];
+  concepts: Array<ConceptInfo> = [];
   conceptsCache: Array<ConceptCacheSet> = [];
   completedDomainSearches: Array<Domain> = [];
   placeholderValue = '';
   vocabularies: Array<VocabularyCountSelected> = [];
   wsNamespace: string;
   wsId: string;
+  selectConceptMap = new Map<number, ConceptInfo>();
```

Comment:
nit: this is only used to check for presence, so a Set of IDs would have sufficed.

Reply:
L361 is n^2 now so would prefer a set. Feel free to build it locally at L360 if you don't need it elsewhere (one of my other comments suggested this)

Reply:
Ok, think I got it.